spot-contracts/contracts/RolloverVault.sol

```diff
@@ -433,13 +427,15 @@ contract RolloverVault is
         // transfer underlying in
         underlying.safeTransferFrom(_msgSender(), address(this), underlyingAmtIn);
 
-        if (execRecover) {
-            // redeem tranches to maximize underlying balance for minting
-            recover();
-        }
-
         // tranche and mint perps as needed
-        _trancheAndMintPerps(perp, perpPrice, perpAmtOut + perpAmtToBurn);
+        {
```

Comment:
Any technical reason for the extra scope block or purely for style?

In c++ this actually pushes another frame on the stack, which is useful in some cases, but not sure about solidity.

Reply:
the compiler imposes a limit on the number of variable definitions within a scoped block